awx/main/models/unified_jobs.py

```diff
@@ -233,7 +233,7 @@ def _perform_unique_checks(self, unique_checks):
                     model_class = self.get_real_concrete_instance_class()
                     errors.setdefault(new_key, []).append(self.unique_error_message(model_class, msg))
             errors[key] = [x for x in msgs if not isinstance(x, (list, tuple))]
-        for key, msgs in errors.items():
+        for key, msgs in errors.copy().items():
```

Comment:
It feels like this change is papering over the fact that there is probably a more straightforward way of building up the error dict.